com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/ZigBeeNetworkManager.java

```diff
@@ -1122,6 +1136,14 @@ public void run() {
                     } else {
                         logger.debug("{}: No successful response received to leave command (status code {})",
                                 leaveAddress, response.getStatusCode());
+                        if (forceNodeRemoval) {
+                            ZigBeeNode node = getNode(leaveAddress);
```

Comment:
I was wondering if this could be consolidated iwth the above check - something like -:

```
                    if (response.getStatusCode() == 0 || forceNodeRemoval) {
```

This would just avoid the (admittedly relatively minor) duplication as in both cases we call ```removeNode()```

Reply:
I see it now, thanks for adding the comment again.

I actually did not include the `forceNodeRemoval` case in the `response.getStatusCode() == 0` case, so that one can log explicitly that the node was "force-removed".

Looking at the code once more now, I think that one could maybe consolidate by pulling the `node = getNode(leaveAddress)` out of the `if` block and put it directly after the `sendTransaction(...,...).get()`, so that one does not have to get the node twice - but let me take a look at this tomorrow when I am at a bigger screen.